akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2Multiplexer.scala

```diff
@@ -154,9 +175,16 @@ private[http2] trait Http2MultiplexerSupport { logic: GraphStageLogic with Stage
             info.closeStream()
             outStreams.remove(sub.streamId)
           } else {
-            val subIn = new SubSinkInlet[ByteString](s"substream-in-${sub.streamId}")
-            info.registerIncomingData(subIn)
-            sub.data.runWith(subIn.sink)(subFusingMaterializer)
+            sub match {
```

Comment:
When the type params are gone, I think no match is even necessary any more. (I think there's still value in keeping the two subclasses of Http2SubStream).

Reply:
Hmm, I don't think `sub.data.runWith(subIn.sink)` will be allowed when `subIn` is a `SubSinkInlet[_]` and `data` is a `Source[_, Any]`, since the compiler has no way to verify those `_`'s line up, right?

Reply:
This seems to work:

```scala
val subIn = new SubSinkInlet[Any](s"substream-in-${sub.streamId}")
info.registerIncomingData(subIn)
sub.data.runWith(subIn.sink)(subFusingMaterializer)
```